src/tools/miri/src/machine.rs

```diff
@@ -65,12 +71,16 @@ pub struct FrameExtra<'tcx> {
     /// optimization.
     /// This is used by `MiriMachine::current_span` and `MiriMachine::caller_span`
     pub is_user_relevant: bool,
+
+    /// Indicates which [`AllocId`] for a [`mir::Const`] this frame will use. This choice is fixed
+    /// per-frame so that within a call, a const will have a stable address.
```

Comment:
```suggestion
    /// We have a cache for the mapping from [`mir::Const`] to resulting [`AllocId`].
    /// However, we don't want all frames to always get the same result, so we insert
    /// an additional bit of "salt" into the cache key. This salt is fixed per-frame
    /// so that within a call, a const will have a stable address
```
Then please also rename `frame_id` to `salt`. (It's already in` FrameExtra` so no need for the `frame_` prefix.)

I'm open to other names, but "id" is not a good choice IMO since the point isn't to identify the frame.